
syntax = "proto3";

package telemetry;

service gRPCMdtDialout {
  rpc MdtDialout(stream Telemetry) returns (Response) {}
}

message Telemetry {
  string node_id_str = 1;
  uint64 subscription_id = 2;
  string sensor_path = 3;
  string encoding_path = 4;
  string model_version = 5;
  uint64 collection_id = 6;
  uint64 collection_start_time = 7;
  uint64 msg_timestamp = 8;
  uint64 collection_end_time = 9;
  uint64 msg_sequence_number = 10;
  bytes data_gpbkv = 11;
  bytes data_gpb = 12;
  string data_json = 13;
  uint64 data_str_len = 14;
  string source_ip = 15;
  string target_ip = 16;
}

message Response {
  string message = 1;
}
